cpmpy/solvers/scip.py

```diff
@@ -0,0 +1,419 @@
+
+                elif lhs.name == 'mul':
+                    assert len(lhs.args) == 2, "scip only supports multiplication with 2 variables"
+                    a, b = self.solver_vars(lhs.args)
+                    #self.scip_model.setParam("NonConvex", 2)  # gurobi leftover
+                    self.scip_model.addCons(a * b == sciprhs)
+
+                elif lhs.name == 'div':
+                    assert is_num(lhs.args[1]), "scip only supports division by constants"
+                    a, b = self.solver_vars(lhs.args)
+                    self.scip_model.addCons(a / b == sciprhs)
+
+                else:
+                    raise NotImplementedError(
+                        "Not a known supported scip comparison '{}' {}".format(lhs.name, cpm_expr))
+
+                    # SCIP does have 'addConsAnd', 'addConsOr', 'addConsXor', 'addConsSOS2' #TODO?
+            else:
+                raise NotImplementedError(
+                "Not a known supported scip comparison '{}' {}".format(lhs.name, cpm_expr))
+
+        elif isinstance(cpm_expr, Operator) and cpm_expr.name == "->":
+            # Indicator constraints
+            # Takes form bvar -> sum(x,y,z) >= rvar
+            cond, sub_expr = cpm_expr.args
+            assert isinstance(cond, _BoolVarImpl), f"Implication constraint {cpm_expr} must have BoolVar as lhs"
+            assert isinstance(sub_expr, Comparison), "Implication must have linear constraints on right hand side"
+
+            lhs, rhs = sub_expr.args
+            assert isinstance(lhs, _NumVarImpl) or lhs.name == "sum" or lhs.name == "wsum", f"Unknown linear expression {lhs} on right side of indicator constraint: {cpm_expr}"
+
+            if sub_expr.name == "<=":  # XXX, docs say only this one?
+                lin_expr = self._make_numexpr(lhs)
+                if isinstance(cond, NegBoolView):
+                    self.scip_model.addConsIndicator(lin_expr <= self.solver_var(rhs),
+                                                     binvar=self.solver_var(cond._bv), activeone=False)
+                else:
+                    self.scip_model.addConsIndicator(lin_expr <= self.solver_var(rhs),
+                                                     binvar=self.solver_var(cond), activeone=True)
+
+            elif sub_expr.name == ">=": # change sign
+                if lhs.name == "sum":
+                    lhs = Operator("wsum", [[-1]*len(lhs.args), lhs.args])
+                elif lhs.name == "wsum":
+                    lhs = Operator("wsum", [[-w for w in lhs.args[0]], lhs.args[1]])
+                else:
+                    lhs = -lhs
+                self += cond.implies(lhs <= -rhs)
+
+            elif sub_expr.name == "==": # split into <= and >=
+                self += [cond.implies(lhs <= rhs), cond.implies(lhs >= rhs)]
+            else:
+                raise Exception(f"Unknown linear expression {sub_expr} name")
+
+        # True or False
+        elif isinstance(cpm_expr, BoolVal):
```

Comment:
Don't really understand what this constraint means, always struggle a bit with the CP language. If you can explain then I can see if there's a more natural MIP approach.

Reply:
This is a very uncommon case where a user posts the constraint "False" to the solver, in essence making the solver status trivialy infeasible. 
We run into this case sometimes when our expressions are simplified to "False"